### PR TITLE
build: switch benchpress NPM output to APF with ESM 

### PR DESCRIPTION
**Pull request by devversion**
See individual commits

### CODE REVIEW

**Comment by devversion** [comment]
This PR in https://github.com/angular/dev-infra/pull/368 shows that this would work as expected, with minimal changes to our benchmark setup. It's a much larger undetaking to refactor all of benchpress, or build an alternative that works with CJS. Running e2e tests with ESM seems like the best long-term solution but that is a larger undertaking.

File: packages/benchpress/package.json [hunk 1]
```diff
@@ -1,9 +1,7 @@
 {
   "name": "@angular/benchpress",
-  "version": "0.2.2",
+  "version": "0.2.3",
```
Comment (josephperrott) [inline packages/benchpress/package.json:3]: As @gkalpak pointed out for that last release we might need to do a minor bump.  With the format change, I think that a minor bump might make sense here for us.

Comment (devversion) [reply packages/benchpress/package.json]: Updated to a minor. Also increased the peer dependency as discussed, allowing for v14 stable/next releases

**Review by josephperrott — APPROVED** [review]
LGTM

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit 53a4a03144a290eac9d46b0756d13720e0ada3e9.

**Pull request closed without merging**